crates/prune/src/pruner.rs

```diff
@@ -265,6 +306,7 @@ impl<DB: Database> Pruner<DB> {
 
 #[cfg(test)]
 mod tests {
```

Comment:
test nit: it'd be ideal to test the limiter logic by introducing a custom `TestSegment: Segment` that either hits the units deletion limit, or the timeout.

Reply:
will save this for last :)

Reply:
maybe out of scope? since storage and account segments are bundled into one type atm in test db, improving the test type system further can be an issue with both these tasks, wdyt?